server.go

```diff
@@ -1541,6 +1555,73 @@ func (c *createMeasurementsIfNotExistsCommand) addFieldIfNotExists(measurement,
 	return nil
 }
 
+func (s *Server) applyDropSeries(m *messaging.Message) error {
+	var c dropSeriesCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	database := s.databases[c.Database]
+	if database == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// Remove from metastore.
+	for _, seriesID := range c.SeriesIDs {
+		series := database.series[seriesID]
+		if series == nil {
+			return fmt.Errorf("series not found for id %d", seriesID)
+		}
+		m := series.measurement
+		if m == nil {
+			return fmt.Errorf("measurement not found for series %d", seriesID)
+		}
+		err := s.meta.mustUpdate(func(tx *metatx) error { return tx.dropSeries(c.Database, m.Name, seriesID) })
+		if err != nil {
+			return err
+		}
+	}
+
+	// Remove shard data
+	for _, rp := range s.databases[c.Database].policies {
```

Comment:
Will do.  Have you done something similar to this already, or will this be the first time we are writing a larger transaction?